### PR TITLE
Port some conveniences from @types/ember__test-helpers package

### PR DESCRIPTION
**Pull request by gitKrystan**
I compared the `@types/ember__test-helpers` package to the public API of this package to ensure we were:
1. exporting the same things (aside from some of the missing exports I put in other PRs)
2. not losing useful type info (e.g. the `find` overloads)

In the process I noticed a couple of minor issues w/ the types and fixed those as well.

LMK if you'd prefer I broke this up.

### CODE REVIEW

File: addon-test-support/@ember/test-helpers/dom/-get-element.ts
Comment (chriskrycho) [inline addon-test-support/@ember/test-helpers/dom/-get-element.ts:10]: I recognize these came from the DT types, but… this is terrible and we should not perpetuate its terribleness.

```suggestion
export default function getElement(selector: string): Element | null;
```

This change prevents people from “casting” it implicitly by writing `getElement<HTMLDivElement>()`. Although that's *convenient*, especially in tests, it's also quite misleading.

File: addon-test-support/@ember/test-helpers/dom/find.ts
Comment (chriskrycho) [inline addon-test-support/@ember/test-helpers/dom/find.ts:12]: Same here!

```suggestion
export default function find(selector: string): Element | null;
```

File: addon-test-support/@ember/test-helpers/dom/trigger-event.ts
Comment (chriskrycho) [inline addon-test-support/@ember/test-helpers/dom/trigger-event.ts:60]: 1. 💙 
2. In the future, we should see if we can make this more specific, but it shouldn't block this change. Making it more accurate will "just" be a bug fix.

File: type-tests/api.ts
Comment (chriskrycho) [inline type-tests/api.ts:53]: No harm done here, but for future reference: I try to keep things like these ordering changes separate from any *meaningful* changes—I just pull them into a separate commit, using `git add --patch` (though I *always* do that via a GUI because I find the text interface for it terrible)—so that it's clear what the meaningful change was. Here, for example, it took some extra reading and comparison to figure out that `RenderingTestContext` was the key change in this set.

**Review by chriskrycho — COMMENTED** [review]
One tweak requested!

File: addon-test-support/@ember/test-helpers/dom/trigger-event.ts
Comment (gitKrystan) [reply addon-test-support/@ember/test-helpers/dom/trigger-event.ts:60]: Yeah, I did a deep dive on what the types should be, and `options` is just passed around willy nilly to a bunch of functions that take `any`, so figuring out the actual type will be a Project™️. I figured `Record<string, unknown>` was at least a little better than `object` and more consistent with another `options` type you specified for a similar test helper.

Comment (gitKrystan) [reply addon-test-support/@ember/test-helpers/dom/trigger-event.ts:60]: I could do something like `Event & Record<string, unknown>` bc I'm pretty sure it's mostly used for specifying details on the `Event` to be triggered, and I think that would at least give some hints about the types for things that exist on an `Event`.

File: type-tests/api.ts
Comment (gitKrystan) [reply type-tests/api.ts:53]: I'll just un-sort for now.

Re `RenderingTestContext`, should it be made public to consumers of this library? Currently it's not actually exported, so this import doesn't even work. I fix that in https://github.com/emberjs/ember-test-helpers/pull/1286, where the same issue came up for different reasons.

File: addon-test-support/@ember/test-helpers/dom/-get-element.ts
Comment (gitKrystan) [inline addon-test-support/@ember/test-helpers/dom/-get-element.ts:12]: These types started as copy-pasta from the built-in `querySelector` types, which look like this:

```ts
    /** Returns the first element that is a descendant of node that matches selectors. */
    querySelector<K extends keyof HTMLElementTagNameMap>(selectors: K): HTMLElementTagNameMap[K] | null;
    querySelector<K extends keyof SVGElementTagNameMap>(selectors: K): SVGElementTagNameMap[K] | null;
    querySelector<E extends Element = Element>(selectors: string): E | null;
```

...but as @chriskrycho pointed out [here](https://github.com/emberjs/ember-test-helpers/pull/1287#discussion_r1049755184), those types aren't that great.

Since we're diverging from them a bit by removing the generic from the `target: string` overload, I decided to also fix a type bug from those types, which is this:

`HTMLElementTagNameMap` and `SVGElementTagNameMap` have some overlap in keys, e.g. `a` is a valid tag in both. In the built-in `querySelector` types, the return type for `target: 'a'` would always be an `HTMLAnchorElement`, when the actual value might actually be a `SVGAElement`. So I added a third overload to account for that overlap. You can see the results in the type-tests.

File: addon-test-support/@ember/test-helpers/dom/trigger-event.ts
Comment (gitKrystan) [reply addon-test-support/@ember/test-helpers/dom/trigger-event.ts:60]: Going to open a separate PR w/ the deep dive since I couldn't help myself. 😂

File: type-tests/api.ts
Comment (chriskrycho) [reply type-tests/api.ts]: Per my note on #1286, I think exporting it is great. 👍🏼

File: addon-test-support/@ember/test-helpers/dom/find-all.ts
Comment (chriskrycho) [inline addon-test-support/@ember/test-helpers/dom/find-all.ts:10]: While this works, I think it'd be slightly clearer to directly index the types:

```suggestion
>(selector: K): Array<HTMLElementTagNameMap[K] | SVGElementTagNameMap[K]>;
```

File: addon-test-support/@ember/test-helpers/dom/find.ts
Comment (chriskrycho) [inline addon-test-support/@ember/test-helpers/dom/find.ts:6]: Same thing here:

```suggestion
>(selector: K): HTMLElementTagNameMap[K] | SVGElementTagNameMap[K] | null;
```

File: addon-test-support/@ember/test-helpers/dom/trigger-event.ts
Comment (chriskrycho) [reply addon-test-support/@ember/test-helpers/dom/trigger-event.ts:60]: Hahaha I love it. Feels very https://xkcd.com/356/ 😂 "A Project™" sounds exactly right, though.

**Review by chriskrycho — CHANGES_REQUESTED** [review]
Noted one further pair of stylistic tweaks we should make, but looks good otherwise!

**Review by chriskrycho — APPROVED** [review]
I applied those tweaks via GH's review UI, so it's now ready to go. Thank you!

**Comment by gitKrystan** [comment]
Found a tiny bug in this one. Will fix w/o force pushing ;-)

**Comment by chriskrycho** [comment]
All right, I fixed a merge conflict of my own creation, will merge once this is ✅ on CI!

**Comment by gitKrystan** [comment]
Should be all gouda now.

**Pull request merged**
